src/Compiler/TypedTree/TypedTree.fs

```diff
@@ -4556,28 +4556,37 @@ type TupInfo =
 type Measure = 
 
     /// A variable unit-of-measure
-    | Var of typar: Typar
+    | Var of typar: Typar * range: range
 
     /// A constant, leaf unit-of-measure such as 'kg' or 'm'
-    | Const of tyconRef: TyconRef
+    | Const of tyconRef: TyconRef * range: range
 
     /// A product of two units of measure
-    | Prod of measure1: Measure * measure2: Measure
+    | Prod of measure1: Measure * measure2: Measure * range: range
```

Comment:
The newly added code in TypedTreeOps is assigning range0 to the range anyway, is there a codepath assigning a meaningful value to it (and different from measure1/2.Range) ?